## Primero leemos el corpus que de antemano generamos:

In [2]:
import pandas as pd
df = pd.read_csv('/Users/user/Tesis/escritorio_anterior/MICAI_2/negativos_2/+bigrama_neg_2.csv',
                     header=0, sep=',', names=['id', 'content', 'label'])
df

id  \
0     AEG_Electrolux_60840_Lavamat__Opinion_1506705.txt   
1     AEG_Electrolux_62610_Lavamat__Opinion_2000923.txt   
2          AEG_Electrolux_L14800VI__Opinion_2005396.txt   
3           AEG_Electrolux_L6227FL__Opinion_2140710.txt   
4          AEG_Electrolux_L62280FL__Opinion_2151025.txt   
5          AEG_Electrolux_L62642VI__Opinion_1995158.txt   
6          AEG_Electrolux_L62642VI__Opinion_1995791.txt   
7          AEG_Electrolux_L70470FL__Opinion_2085167.txt   
8            AEG_Electrolux_L74650__Opinion_1856967.txt   
9         AEG_Electrolux_L85275XFL__Opinion_2138100.txt   
10        AEG_Electrolux_L_14950_A__Opinion_2103767.txt   
11          AEG_Electrolux_L_47185__Opinion_1492496.txt   
12          AEG_Electrolux_L_47185__Opinion_1818711.txt   
13          AEG_Electrolux_L_47186__Opinion_1370192.txt   
14        AEG_Electrolux_L_47330_A__Opinion_2087353.txt   
15           AEG_Electrolux_L_5214__Opinion_1437024.txt   
16          AEG_Electrolux_L_62825__Opinion_1469992.txt   
17       AEG_Electrolux_L_70270_FL__Opinion_2090317.txt   
18       AEG_Electrolux_L_70270_FL__Opinion_2143750.txt   
19       AEG_Electrolux_L_70270_FL__Opinion_2150873.txt   
20       AEG_Electrolux_L_70270_FL__Opinion_2152723.txt   
21          AEG_Electrolux_L_70810__Opinion_1263244.txt   
22          AEG_Electrolux_L_70810__Opinion_1277946.txt   
23          AEG_Electrolux_L_70810__Opinion_1284085.txt   
24          AEG_Electrolux_L_70810__Opinion_1372777.txt   
25          AEG_Electrolux_L_70810__Opinion_1848780.txt   
26          AEG_Electrolux_L_70850__Opinion_1448444.txt   
27          AEG_Electrolux_L_70850__Opinion_2007327.txt   
28         AEG_Electrolux_L_725500__Opinion_1791610.txt   
29         AEG_Electrolux_L_725500__Opinion_1794746.txt   
...                                                 ...   
2568            Zanussi_ZWH_6101_P__Opinion_1999819.txt   
2569            Zanussi_ZWH_6101_P__Opinion_2001127.txt   
2570            Zanussi_ZWH_6101_P__Opinion_2001296.txt   
2571            Zanussi_ZWH_6101_P__Opinion_2001662.txt   
2572            Zanussi_ZWH_6101_P__Opinion_2006735.txt   
2573            Zanussi_ZWH_6101_P__Opinion_2007532.txt   
2574            Zanussi_ZWH_6101_P__Opinion_2009115.txt   
2575            Zanussi_ZWH_6101_P__Opinion_2026016.txt   
2576            Zanussi_ZWH_6101_P__Opinion_2027055.txt   
2577           Zanussi_ZWH_6102_PS__Opinion_2138338.txt   
2578              Zanussi_ZWH_6120__Opinion_1981065.txt   
2579              Zanussi_ZWH_6120__Opinion_2136981.txt   
2580              Zanussi_ZWH_6125__Opinion_1701351.txt   
2581            Zanussi_ZWH_7120_P__Opinion_2005676.txt   
2582            Zanussi_ZWH_7120_P__Opinion_2132623.txt   
2583            Zanussi_ZWH_7125_X__Opinion_2083788.txt   
2584            Zanussi_ZWH_7125_X__Opinion_2127841.txt   
2585              Zanussi_ZWK_5100__Opinion_1526854.txt   
2586              Zanussi_ZWK_6120__Opinion_1613313.txt   
2587              Zanussi_ZWN_5105__Opinion_2122850.txt   
2588              Zanussi_ZWN_6105__Opinion_2004500.txt   
2589              Zanussi_ZWN_6120__Opinion_2084720.txt   
2590             Zanussi_ZWN_7100L__Opinion_2093055.txt   
2591            Zanussi_ZWN_7120_L__Opinion_2131960.txt   
2592              Zanussi_ZWN_7125__Opinion_2093504.txt   
2593              Zanussi_ZWQ_5100__Opinion_2138972.txt   
2594              Zanussi_ZWQ_5109__Opinion_2094551.txt   
2595               Zanussi_ZWQ_598__Opinion_2004693.txt   
2596               Zanussi_ZWT_260__Opinion_1091527.txt   
2597               Zanussi_ZWT_260__Opinion_1157262.txt   

                                                content  label  
0     programa especial ropa sensible planchado fáci...      5  
1     carga superior punto medio color blanco carga ...      4  
2     carga superior forma frontal programa majo efi...      5  
3     modelo electrolux muy fácil altamente satisfac...      4  
4     clasificación energético sonido bajo conversac...      4  
5   

In [3]:
from sklearn.feature_extraction.text import TfidfVectorizer
tfidf_vect = TfidfVectorizer(use_idf=True, smooth_idf=True,
                             sublinear_tf=False, ngram_range=(2,2))

X = tfidf_vect.fit_transform(df['content'].values)
y = df['label'].values

## Ahora hacemos la validación cruzada en la que el Dr. Rosso tenia duda:

Primero contamos las opiniones:

In [4]:
df.count()

id         2598
content    2598
label      2598
dtype: int64

Ahora haremos la validacion cruzada con el 33% para prueba y el 70% para entrenar en cada fold. A 10 folds, imprimimos los indices de las opiniones:

In [7]:
from sklearn import cross_validation

#Usamos una estrategia Cross validation ShuffleSplit:
#(http://scikit-learn.org/stable/modules/generated/sklearn.cross_validation.ShuffleSplit.html
#sklearn.cross_validation.ShuffleSplit)
ss = cross_validation.ShuffleSplit(n = len(y), n_iter=10, test_size=0.30,random_state=0)


import numpy
numpy.set_printoptions(threshold=numpy.nan)

for train_index, test_index in ss:
    print "\n \n Entrenamiento 30% de 2598:\n", train_index, \
    "\n \n Prueba 70% de 2598:\n", test_index
    X_train, X_test = X[train_index], X[test_index]
    y_train, y_test = y[train_index], y[test_index]



 
 Entrenamiento 30% de 2598:
[ 483   80 1593  270 1709 1232 1602  475  532 1881  299 2330 1922  485  233
 2484 1759 2548 2160  287 2427  733  512  766  895 2499  655 2114  553 1228
  279 1078  253  663  250  169 1374 1405 2343 2230  949 1928  962  847   31
 2433 1449 1621 1387  900 1912 2267 2503 1900  868  926 2186 1675  359  369
  794  309 1850 1246 1798  818 1114 1548 1326 2170  966 1452   87 1199 1749
 2166  141   64 2580 2268 2393 1852  521 2125 1719 2123 2132   77 1555  393
 1388  262 2200  923  240  721 1338 2539 1175 1325 2064  810 1005 1697  799
 1855 1767  381 1486 1878 2033 1482  964 2303 2311 1339  491  106 1622   85
  202  836   15  474 1996  118 1983 2590 1023  502 1317  701 1091 2156  654
 1846 2563  633  629 2341 1222 1526  600 1347 1161 1937 1280  825  793 1594
  791 1013 1426  883  310  735  187  298  636 1070  775 1431 2110 2573  415
  516  917  971 2196 1271  564  294 1894 1038 2047 1380 1892  572 2518  267
 2386 1273 1875  144 1780 1068 2055  260  806  781  832 1

## Imprimimos el tamaño de cada fold o split

In [6]:
print 'tamaño del train para cada fold es:',train_index.size
print 'tamaño del test para cada fold es:',test_index.size

tamaño del train para cada fold es: 1818
tamaño del test para cada fold es: 780


Luego teniamos **2598 opiniones**. **El 70% de las 2589 opiniones es 1818.6** y el **30% de las 2598 opiniones es 779.4**. Por la celda anterior se verifica. 